daemon/volumes.go

```diff
@@ -119,8 +119,18 @@ func (container *Container) VolumePaths() map[string]struct{} {
 }
 
 func (container *Container) registerVolumes() {
-	for _, mnt := range container.VolumeMounts() {
-		mnt.volume.AddContainer(container.ID)
+	for path := range container.VolumePaths() {
+		if v := container.daemon.volumes.Get(path); v != nil {
+			v.AddContainer(container.ID)
+		} else {
+			// if container was created with an old daemon, this volume may not be registered
+			v, err := container.daemon.volumes.FindOrCreateVolume(path, container.VolumesRW[path])
```

Comment:
It would be pre 1.3 daemons, so not that old.

I'll update this VolumesRW thing to make sure we don't panic here.  It _shouldn't_ be nil, but it could be.